crates/clarirs_num/src/bitvec/conversion.rs

```diff
@@ -1,70 +1,82 @@
 use num_bigint::BigUint;
 use smallvec::SmallVec;
+use std::convert::Infallible;
+use std::convert::TryFrom;
 
 use super::{BitVec, BitVecError};
 
-impl From<i8> for BitVec {
-    fn from(value: i8) -> Self {
+impl TryFrom<i8> for BitVec {
+    type Error = BitVecError;
+    fn try_from(value: i8) -> Result<BitVec, BitVecError> {
         BitVec::new(SmallVec::from_slice(&[value as u64]), 8)
     }
 }
 
-impl From<u8> for BitVec {
-    fn from(value: u8) -> Self {
+impl TryFrom<u8> for BitVec {
```

Comment:
For these cases, what is the error that could occur here? `BitVec::new` returns a Result, but I think since we know none of these could overflow, we should be able to get away with using `.expect("a u8 was large than 8 bits?")` and similar